src/plugins/data/public/search/aggs/agg_config.ts

```diff
@@ -27,18 +28,20 @@ import { ISearchSource } from '../search_source';
 import { FieldFormatsContentType, KBN_FIELD_TYPES } from '../../../common';
 import { FieldFormatsStart } from '../../field_formats';
 
-export interface AggConfigOptions {
-  type: IAggType;
+export interface AggConfigJson {
+  type: string;
   enabled?: boolean;
   id?: string;
-  params?: Record<string, any>;
+  params?: Record<string, any>; // TODO: this should enforce serializable state
   schema?: string;
 }
 
 export interface AggConfigDependencies {
   fieldFormats: FieldFormatsStart;
 }
 
+export type AggConfigOptions = Assign<AggConfigJson, { type: IAggType }>;
```

Comment:
hmm, will the type be the only different thing ? what about params, don't we instantiate anything there ?

Reply:
yeah it seems params are serialized as well

Reply:
yeah AFAIK params come in to the constructor serialized, and need to then be deserialized in `setParams`

`getParam` will return the deserialized params, and `toJSON` should re-serialize everything again.

Reply:
Now that I've updated `AggConfigSerialized`, it should be enforced via TS that any params coming into the constructor are serializable. They are then deserialized and added to the AggConfig instance.